crates/router/src/core/blocklist/transformers.rs

```diff
@@ -11,3 +29,195 @@ impl ForeignFrom<storage::Blocklist> for blocklist::AddToBlocklistResponse {
         }
     }
 }
+
+async fn generate_fingerprint_request<'a>(
+    #[cfg(not(feature = "aws_kms"))] jwekey: &settings::Jwekey,
+    #[cfg(feature = "aws_kms")] jwekey: &settings::ActiveKmsSecrets,
+    locker: &settings::Locker,
+    payload: &blocklist::GenerateFingerprintRequest,
+    locker_choice: api_enums::LockerChoice,
+) -> CustomResult<services::Request, errors::VaultError> {
+    let payload = utils::Encode::<blocklist::GenerateFingerprintRequest>::encode_to_vec(&payload)
+        .change_context(errors::VaultError::RequestEncodingFailed)?;
+
+    #[cfg(feature = "aws_kms")]
+    let private_key = jwekey.jwekey.peek().vault_private_key.as_bytes();
+
+    #[cfg(not(feature = "aws_kms"))]
+    let private_key = jwekey.vault_private_key.as_bytes();
+
+    let jws = encryption::jws_sign_payload(&payload, &locker.locker_signing_key_id, private_key)
+        .await
+        .change_context(errors::VaultError::RequestEncodingFailed)?;
+
+    let jwe_payload = generate_jwe_payload_for_request(jwekey, &jws, locker_choice).await?;
+    let mut url = match locker_choice {
+        api_enums::LockerChoice::Basilisk => locker.host.to_owned(),
+        api_enums::LockerChoice::Tartarus => locker.host_rs.to_owned(),
+    };
+    url.push_str(LOCKER_API_URL);
+    let mut request = services::Request::new(services::Method::Post, &url);
+    request.add_header(headers::CONTENT_TYPE, "application/json".into());
+    request.set_body(RequestContent::Json(Box::new(jwe_payload)));
+    Ok(request)
+}
+
+async fn generate_jwe_payload_for_request(
```

Comment:
can we not re-implement the same functions that are already existsing? instead, make the functions that we already have and are using in stored cards generic and use them here

Reply:
I agree regarding the generation logic, I was specifically talking about the encryption logic. As, it is same for every case, it would be better and more maintainable if centralized.

It's ok to have it in a separate PR, but please create a Issue and link it here, as its better done quickly, as the code is ever-changing

Reply:
Sure the [issue](https://github.com/juspay/hyperswitch/issues/3697) is created and will be addressed asap.